### PR TITLE
DEV: Introduce a `Service::ActionBase` class for service actions

### PR DESCRIPTION
**Pull request by Flink**
This will help to enforce a consistent pattern for creating service actions.

This patch also namespaces actions and policies, making everything related to a service available directly in `app/services/<concept-name>`, making things more consistent at that level too.

### CODE REVIEW

File: plugins/chat/app/services/chat/channel/policy/message_creation.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # frozen_string_literal: true
 
-class Chat::Channel::MessageCreationPolicy < PolicyBase
+class Chat::Channel::Policy::MessageCreation < Service::PolicyBase
```
Comment (Flink) [inline plugins/chat/app/services/chat/channel/policy/message_creation.rb:3]: We should probably apply that `Chat::Channel` namespace to related services, I’ll do that in another PR :sweat_smile:

**Review by davidtaylorhq — APPROVED** [review]
